### PR TITLE
WIP: Docs

### PR DESCRIPTION
**Pull request by CFGrote**
This PR adds several improvements to the documentation. Mostly following PR #48 plus better structure in the Reference Manual

### CODE REVIEW

**Comment by JunCEEE** [comment]
This error message: `ValueError: len(output_keys) = 1 is not equal to len(output_data_types) = 1`
from [here](https://github.com/PaNOSC-ViNYL/libpyvinyl/blob/bcc26328cd05a51a39b570c10c1e1bea1ac3d09c/doc/source/include/notebooks/example-01.ipynb)
is supposed to be `ValueError: len(output_keys) = 1 is not equal to len(output_data_types) = 0`.
It' fixed here:
https://github.com/PaNOSC-ViNYL/libpyvinyl/blob/2a226f56810f711a16a24dc4f202fe8f99d763eb/libpyvinyl/BaseCalculator.py#L133-L138

File: README.md [hunk 1]
```diff
@@ -12,22 +12,119 @@ While `Parameters` is a pure state engine, i.e. it's sole purpose is to encapsul
...
+
+### Developers
+We provide a requirements file for developers in *requirements/dev.txt*.   
+
+```
+$> cd requirements  
+```
+
+```
+$> conda install --file dev.txt
```
Comment (JunCEEE) [inline README.md:77]: `conda` doesn't support a file reference inside a file. One will get an error: 
```
CondaValueError: could not parse '-r prod.txt' in: dev.txt
```
I suggest removing this line.

Comment (shervin86) [reply README.md:77]: I would then suggest to change the documentation as follows:
```suggestion
$> conda install --file requirements/dev.txt --file requirements/prod.txt
```
and for pip:
```
pip install -r requirements/prod.txt -r requirements/dev.txt
```
and to remove the `-r prod.txt` line from the dev.txt file

**Comment by CFGrote** [comment]
can we merge this asap please?

**Review by shervin86 — APPROVED** [review]

**Comment by shervin86** [comment]
can we delete the `docs` branch now?

**Comment by JunCEEE** [comment]
Yes

**Pull request merged**